src/main/java/seedu/address/storage/XmlAddressBookStorage.java

```diff
@@ -74,4 +85,63 @@ public void saveAddressBook(ReadOnlyAddressBook addressBook, String filePath) th
         XmlFileStorage.saveDataToFile(file, new XmlSerializableAddressBook(addressBook));
     }
 
+    @Override
+    public void backupAddressBook() throws IOException {
+        try {
+            Optional<ReadOnlyAddressBook> addressBookOptional;
+            ReadOnlyAddressBook initialData;
+            addressBookOptional = readAddressBook();
+            if (!addressBookOptional.isPresent()) {
+                logger.info("No backup will be made as data file does not exist");
+            } else {
+                initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
+                saveAddressBook(initialData, getBackupAddressBookFilePath());
+            }
+        } catch (DataConversionException e) {
+            logger.info("No backup will be made due to data file being in incorrect format");
+        } catch (IOException e) {
+            logger.info("No backup will be made due to problems while reading from file");
+        }
+    }
+
+    @Override
+    public ReadOnlyAddressBook getBestAvailableAddressBook() {
```

Comment:
Missing header comment. All non trivial methods should have javadoc format header comments.

Reply:
Noted. Necessary changes have been made by the group.